### PR TITLE
MULE-12770: Sdk should validate that required parameters can't receiv…

### PR DESCRIPTION
**Pull request by marianogonzalez**
…e null values

### CODE REVIEW

File: modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/resolver/RequiredExpressionParameterValueResolverWrapper.java [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+  private final String expression;
+
+  /**
+   * Creates a new instance
+   *
+   * @param delegate      the wrapped {@link ValueResolver}
+   * @param parameterName the name of the parameter this resolver is associated to
+   * @param expression    the evaluated expression
+   */
+  public RequiredExpressionParameterValueResolverWrapper(ValueResolver<T> delegate,
```
Comment (estebanwasinger) [inline modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/resolver/RequiredExpressionParameterValueResolverWrapper.java]: The name is misleading, you say that is a `RequiredExpressionParameterValueResolverWrapper` but the delegate is not necessary an Expression based `ValueResolver`

File: modules/extensions-spring-support/src/test/java/org/mule/test/module/extension/OperationExecutionTestCase.java [hunk 2]
```diff
@@ -263,6 +264,14 @@ public void operationWithStaticInlinePojoParameter() throws Exception {
   }
 
   @Test
+  public void operationWithRequiredParameterButNullReturningExpression() throws Exception {
+    expectedException.expect(MessagingException.class);
+    expectedException.expectCause(instanceOf(IllegalArgumentException.class));
+    runFlow("knockWithNullDoor");
+  }
+
```
Comment (estebanwasinger) [inline modules/extensions-spring-support/src/test/java/org/mule/test/module/extension/OperationExecutionTestCase.java]: empty line 🙃

File: modules/extensions-spring-support/src/main/java/org/mule/runtime/module/extension/internal/config/dsl/ExtensionDefinitionParser.java [hunk 3]
```diff
@@ -640,15 +641,18 @@ protected ClassLoader getContextClassLoader() {
...
-        ? getExpressionBasedValueResolver(expectedType, (String) value, modelProperties, expectedClass)
-        : getStaticValueResolver(parameterName, expectedType, value, defaultValue, modelProperties, acceptsReferences,
-                                 expectedClass);
+    final Class<Object> expectedClass = ExtensionMetadataTypeUtils.getType(expectedType).orElse(Object.class);
 
+    if (isExpression(value, parser)) {
+      final String expression = (String) value;
+      resolver = getExpressionBasedValueResolver(expectedType, expression, modelProperties, expectedClass);
+      if (required) {
+        resolver = new RequiredExpressionParameterValueResolverWrapper(resolver, parameterName, expression);
```
Comment (estebanwasinger) [inline modules/extensions-spring-support/src/main/java/org/mule/runtime/module/extension/internal/config/dsl/ExtensionDefinitionParser.java]: this "null-safe" wrapper should not be applied for both, the expression and static based resolvers?

Comment (marianogonzalez) [reply modules/extensions-spring-support/src/main/java/org/mule/runtime/module/extension/internal/config/dsl/ExtensionDefinitionParser.java]: how do you do a static one which returns null?

**Pull request merged**